src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,80 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todos: Todo[];
+  onFilter: React.Dispatch<React.SetStateAction<Todo[]>>;
+}
+
+export const TodoFilter: React.FC<Props> = ({ todos, onFilter }) => {
+  const [query, setQuery] = useState('');
+  const [selectedFilter, setSelected] = useState('all');
+
+  useEffect(() => {
+    const filteredTodos = todos
+      .filter(todo => {
+        const lowecasedQuery = query.toLocaleLowerCase();
+        const lowercasedTitle = todo.title.toLocaleLowerCase();
+        const fitsQuery = lowercasedTitle.includes(lowecasedQuery);
+
+        switch (selectedFilter) {
+          case 'active':
+            return !todo.completed && fitsQuery;
+          case 'completed':
+            return todo.completed && fitsQuery;
+          default:
+            return fitsQuery;
+        }
+      });
```

Comment:
and for this you don't need useEffect, you can just use `useMemo`, so you'll avoid creating another `visibleTodos` state